### DISCUSSION TITLE
[Bug]: Infinite Scroll Example does not settle when sdditional slides where added

### DISCUSSION THREAD

**Discussion by elite-benni** [davidjerleke/embla-carousel#791] [Help] [answered]
### Summary

First of all, thanks for the great work, really appreciate it!

I was playing around with the Infinite Scroll example and have an issue. 
After adding additional slides, the `scroll` events won't stop firing and the `settle` event is never called. 

Is there a way to stop that internal scrolling? 

This is reproducable in the infinite Scroll example CodeSandbox. 
I just added a console.log('scroll') to the scroll event. 

After the slides 6-10 are added, scroll fires infinitely, this causes some problems with my implementation. 
Do you have a workaround for this? 

Thanks in advance!

A Fork with scroll log was added: 
[Sandbox](https://codesandbox.io/p/sandbox/embla-carousel-infinite-scroll-vanilla-forked-5n2ttm?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt74in2q0006356l2ko7kc0h%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt74in2q0002356li4ym5qxg%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt74in2q0003356lx7m7syvm%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt74in2q0005356lo26zc5fg%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B49.785967493813125%252C50.214032506186875%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt74in2q0002356li4ym5qxg%2522%253A%257B%2522id%2522%253A%2522clt74in2q0002356li4ym5qxg%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltmswrjn0002356ljybk5tjb%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A39%252C%2522startColumn%2522%253A13%252C%2522endLineNumber%2522%253A39%252C%2522endColumn%2522%253A13%257D%255D%252C%2522filepath%2522%253A%2522%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cltqwxylo0002356lh0h11zxt%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A119%252C%2522startColumn%2522%253A4%252C%2522endLineNumber%2522%253A119%252C%2522endColumn%2522%253A4%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fjs%252FEmblaCarouselInfiniteScroll.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cltqwxylo0002356lh0h11zxt%2522%257D%252C%2522clt74in2q0005356lo26zc5fg%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt74in2q0004356li3yib8we%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clt74in2q0005356lo26zc5fg%2522%252C%2522activeTabId%2522%253A%2522clt74in2q0004356li3yib8we%2522%257D%252C%2522clt74in2q0003356lx7m7syvm%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt74in2q0003356lx7m7syvm%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
Hi @elite-benni,

Thank you for your question. I can confirm that this is a bug. Could you try adding the following [here](https://codesandbox.io/p/sandbox/embla-carousel-infinite-scroll-vanilla-forked-5n2ttm?file=%2Fsrc%2Fjs%2FEmblaCarouselInfiniteScroll.ts):

```tsx
const copyEngineModules: (keyof EngineType)[] = [
  'scrollBody',
  'location',
  'offsetLocation', // <-- Add this
  'target'
]
```

...and let me know if it solves the problem?

Best,
David

**Reply by elite-benni** [reply]
Thank you very much David! 
This seems to work!

**Reply by davidjerleke** [reply]
@elite-benni, if you want to contribute, free to create a PR and add the `'offsetLocation'` line:

- [Here](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel-docs/src/components/Sandbox/Vanilla/SandboxFilesSrc/InfiniteScroll/EmblaCarouselInfiniteScroll.ts#L66-L70).
- And [here](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel-docs/src/components/Sandbox/React/SandboxFilesSrc/InfiniteScroll/EmblaCarousel.tsx#L44-L48).

...which will solve the bug.

Best,
David

**Comment by davidjerleke** [comment]
PR that solves this bug here:

- #797

**Reply by davidjerleke** [reply]
Released in [v8.0.1](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.1).
